src/server.c

```diff
@@ -1801,6 +1801,7 @@ void beforeSleep(struct aeEventLoop *eventLoop) {
         do {
             /* Try to process all the pending IO events. */
             last_processed = processIOThreadsReadDone() + processIOThreadsWriteDone();
+            if (last_processed > 0) server.el_iteration_active = true;
```

Comment:
There are a few more places above 8ncrementing `processed`:

```c
if (ProcessingEventsWhileBlocked) {
        uint64_t processed = 0;
        processed += processIOThreadsReadDone();
        processed += connTypeProcessPendingData();
```

Shouldn't these too make it an active el iteration? Can we simply set it to true if `processed > 0` before the return below, or would that include something we don't want to count?

Reply:
From what I can see, ProcessingEventsWhileBlocked is only set to true while loading an RDB or AOF file and during full sync loading (to reply the -LOADING error to clients) and during slow scripts to allow SCRIPT KILL. In these cases, we get here via the function processEventsWhileBlocked which is most likely called from an active iteration, so if I understand this correctly, we don't need to set `server.el_iteration_active = true` at all when `ProcessingEventsWhileBlocked`. 🤔